### PR TITLE
REF: remove unnecessary PeriodIndex._get_indexer

### PR DESCRIPTION
**Pull request by jbrockmendel**

### CODE REVIEW

File: pandas/core/indexes/period.py [hunk 1]
```diff
@@ -579,14 +569,6 @@ def _get_string_slice(self, key: str):
         except KeyError as err:
             raise KeyError(key) from err
 
-    # ------------------------------------------------------------------------
-
-    def memory_usage(self, deep: bool = False) -> int:
```
Comment (jreback) [inline pandas/core/indexes/period.py:584]: do we have a test that actually exercises PI for memory_usage?

**Review by jreback — APPROVED** [review]

File: pandas/core/indexes/period.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/core/indexes/period.py:584]: yes

**Comment by jreback** [comment]
hmm lots of things are failing here. maybe an ensure_indexer issue?

**Comment by jreback** [comment]
worse: https://dev.azure.com/pandas-dev/pandas/_build/results?buildId=53846&view=logs&j=acc1347f-8235-55b0-95c7-0e2189e61659&t=486f34f1-eda6-516a-fc5d-d8195128dacd

```
>   value = parse_timedelta_string(value)
E   TypeError: Expected unicode, got numpy.str_
```

**Comment by jbrockmendel** [comment]
> hmm lots of things are failing here. maybe an ensure_indexer issue?

i think i switched the order of a np.asarray/to_timedelta call, just tried re-ordering them

**Pull request merged**